### PR TITLE
tests: Write the notification message to different files to avoid race condition in feature_notifications.py

### PR DESCRIPTION
**Pull request by ken2812221**
This PR change the behavior that `feature_notifications.py` would write to different files instead of writing to  the same file to avoid race condition.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
utACK f8b6424a1014e8080f22b39a6d3b9998e85f2f90

**Comment by sdaftuar** [comment]
Could you explain the race condition?

**Comment by ken2812221** [comment]
>Could you explain the race condition?

Bitcoin Core runs the each notification in a new thread, so it would have 10 process writing the same file if it generate 10 blocks immediately in this tests.

**Comment by MarcoFalke** [comment]
Maybe on a second thought the tests should mirror how the notifications are supposed to be used in practice. If we don't support support filesystems that deny access when a lock is taken, as opposed to waiting, then the test shouldn't run on that system.

**Comment by ryanofsky** [comment]
Instead of sleeping between the blocknotify commands, so they can all write to the same file, what about just having them write to different files. E.g. create a `blocknotify/` directory and change `-blocknotify=echo %s >> blocks.txt` to `-blocknotify=echo > blocknotify/%s`

**Comment by promag** [comment]
@ryanofsky suggestion is good to fix the problem.

But I wonder if the we should replace the "call system with user command in a new thread" with a FIFO notification thread?

**Comment by ken2812221** [comment]
@ryanofsky Updated the commit. Now the test create empty files with different name.

File: test/functional/feature_notifications.py [hunk 1]
```diff
@@ -36,54 +39,49 @@ def run_test(self):
         blocks = self.nodes[1].generate(block_count)
 
         # wait at most 10 seconds for expected file size before reading the content
```
Comment (sdaftuar) [inline test/functional/feature_notifications.py]: nit: comment should be updated: `...expected directory size...`

Comment (sdaftuar) [reply test/functional/feature_notifications.py]: also the comments below this one that refer to "file contents" should be "directory contents"

**Review by sdaftuar — COMMENTED** [review]
The last commit looks good to me, modulo the comment nits.  I have not reviewed the first two commits which are from #14007.

**Pull request merged**